internal/flag/flag.go

```diff
@@ -588,3 +591,19 @@ Repeat for each volume in order (comma-separated lists not accepted)
 `,
 	}
 }
+
+// WithExtraArgs derives a context that carries extraArgs from ctx.
+func WithExtraArgs(ctx context.Context, extraArgs []string) context.Context {
+	return context.WithValue(ctx, extraArgsContextKey{}, extraArgs)
+}
+
+// ExtraArgsFromContext returns the extraArgs ctx carries.
+func ExtraArgsFromContext(ctx context.Context) []string {
+	if extraArgs, ok := ctx.Value(extraArgsContextKey{}).([]string); ok {
+		return extraArgs
+	}
+
+	fmt.Printf("unable to get extra args from context\n")
```

Comment:
It seems like error handling should be moved to the caller. Either change the function signature to return a bool for existence:

```go
func ExtraArgsFromContext(ctx context.Context) ([]string, bool)
```

or use `nil` to indicate non-existent extra args and a zero-length `[]string` to indicate empty extra args.

Reply:
First, I merely copied the idea from other places:

```
% git grep 'ctx.Value' | grep ok
internal/appconfig/context.go:	if cfg, ok := ctx.Value(configContextKey).(*Config); ok {
internal/appconfig/context.go:	if name, ok := ctx.Value(nameContextKey).(string); ok {
internal/flag/flag.go:	if extraArgs, ok := ctx.Value(extraArgsContextKey).(*[]string); ok {
internal/metrics/helpers.go:	val, ok := ctx.Value(disableFlushMetricsKey{}).(bool)
```

Are those other cases wrong?

Second, I'm already returning a zero length `[]string` to indicate empty extra args.

My feeling overall is no extra args is not an error, it actually is the normal case.  There is no reason for the caller to care.

Let me know if you disagree.

Finally, the`fmt.Printf` is leftover debug.  I'll remove it.

Reply:
I'll leave it as is as it is consistent with other sections of the code.